utf8proc.c

```diff
@@ -365,9 +365,16 @@ static utf8proc_ssize_t seqindex_write_char_decomposed(utf8proc_uint16_t seqinde
   for (; len >= 0; entry++, len--) {
     utf8proc_int32_t entry_cp = seqindex_decode_entry(&entry);
 
-    written += utf8proc_decompose_char(entry_cp, dst+written,
-      (bufsize > written) ? (bufsize - written) : 0, options,
-    last_boundclass);
+    if (!dst){
+      written += utf8proc_decompose_char(entry_cp, dst,
+        (bufsize > written) ? (bufsize - written) : 0, options,
+      last_boundclass);
+    }else{
+      written += utf8proc_decompose_char(entry_cp, dst+written,
+        (bufsize > written) ? (bufsize - written) : 0, options,
+      last_boundclass);
+    }
+
```

Comment:
Maybe cleaner to combine into a single call with a `dst ? ...` argument
```suggestion
    written += utf8proc_decompose_char(entry_cp, 
      dst ? dst+written : dst,
      (bufsize > written) ? (bufsize - written) : 0, options,
      last_boundclass);
```